R/drive_path.R

```diff
@@ -30,39 +30,44 @@ get_leaf <- function(path = NULL) {
   ## guarantee: we have found at least one item with correct name for
   ## each piece of path
 
-  hits$depth <- match(hits$name, path_pieces)
-  hits <- hits[order(hits$depth), ]
-
   ## leaf candidate(s)
-  leaf_id <- hits$id[hits$depth == d]
+  leaf_id <- hits$id[hits$name == last(path_pieces)]
```

Comment:
I think this is going to generalize better to case where we need to loop backwards over path pieces, until we find one that actually exists.